src/extension/geopkg-output/src/main/java/org/geoserver/geopkg/GeoPkg.java

```diff
@@ -43,12 +43,15 @@ public class GeoPkg {
      */
     public static GeoPackage getGeoPackage(File file) throws IOException {
         SQLiteConfig config = new SQLiteConfig();
-        config.setSharedCache(true);
+        config.setSharedCache(false);
         config.setJournalMode(SQLiteConfig.JournalMode.OFF);
-        config.setPragma(SQLiteConfig.Pragma.SYNCHRONOUS, "OFF");
-        config.setLockingMode(SQLiteConfig.LockingMode.EXCLUSIVE);
+        config.setSynchronous(SQLiteConfig.SynchronousMode.OFF);
+        config.setLockingMode(SQLiteConfig.LockingMode.NORMAL);
+        config.setTempStore(SQLiteConfig.TempStore.MEMORY);
```

Comment:
This worries me too, this class is shared with the WPS process, which can dump massive amounts of data (and well, the WFS output format can too, it will dump to disk before trying to write on the output, so the HTTP timeout won't save it).